### PR TITLE
Make issued_token_type optional to support OAuth2 Client Credential Flow

### PR DESCRIPTION
**Pull request by flyrain**
To fix issue(https://github.com/apache/iceberg-python/issues/463). In short, Client Credential Flow's http response doesn't have the field `issued_token_type`. Make it optional to avoid validation error like this:
```
ValidationError: 1 validation error for TokenResponse
issued_token_type
  Field required [type=missing, input_value={'access_token': 'eyJhbGc... 'token_type': 'bearer'},
input_type=dict]
```
cc @danielcweeks @Fokko @syun64 @RussellSpitzer

### CODE REVIEW

File: pyiceberg/catalog/rest.py
Comment (syun64) [inline pyiceberg/catalog/rest.py:162]: This is a nit, could we use Field annotation like the attributes above?
[Example](https://github.com/apache/iceberg-python/blob/5209874cd4685427520b67b64188f5e9919f2816/pyiceberg/table/snapshots.py#L141)

```suggestion
    expires_in: Optional[int] = Field(default=None)
    issued_token_type: Optional[str] = Field(default=None)
    refresh_token: Optional[str] = Field(default=None)
    scope: Optional[str] = Field(default=None)
```

Comment (flyrain) [reply pyiceberg/catalog/rest.py]: Fixed. Thanks for the suggestion. A curious question: what do we prefer a Field annotation here?

Comment (syun64) [reply pyiceberg/catalog/rest.py]: From my perspective, just styling - it makes it consistent with how we are labeling the other Models in the repository

**Review by syun64 — APPROVED** [review]
Looks good from my side. Let's seek approval from @Fokko @danielcweeks to get this merged in 👍

**Comment by flyrain** [comment]
Thanks @syun64 for the review. We will need at least an approval from committers. cc @Fokko @danielcweeks

File: pyiceberg/catalog/rest.py
Comment (Fokko) [inline pyiceberg/catalog/rest.py:160]: We could also convert this into a literal, since it is an enum in the spec:
```suggestion
    issued_token_type: Optional[Literal['urn:ietf:params:oauth:token-type:access_token', 'urn:ietf:params:oauth:token-type:refresh_token', 'urn:ietf:params:oauth:token-type:id_token', 'urn:ietf:params:oauth:token-type:saml1', 'urn:ietf:params:oauth:token-type:saml2', 'urn:ietf:params:oauth:token-type:jwt'] = Field(default=None)
```

**Review by Fokko — APPROVED** [review]
Thank you @flyrain for fixing this 👍 I wasn't aware of this discrepancy!

Thanks @syun64 for the great review 🙌

**Comment by flyrain** [comment]
Thanks @Fokko for the review!

**Pull request merged**
